lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -253,13 +253,11 @@ export class UserAgentApplication {
         window.msal = this;
 
         const urlHash = window.location.hash;
-        const urlContainsHash = this.urlContainsHash(urlHash);
+        const urlContainsHash = UrlUtils.urlContainsHash(urlHash);
 
         // On the server 302 - Redirect, handle this
-        if (!this.config.framework.isAngular) {
-            if (urlContainsHash) {
-                this.handleAuthenticationResponse(urlHash);
-            }
+        if (!this.config.framework.isAngular && urlContainsHash && !IframeUtils.isInIframe()) {
```

Comment:
May need to specifically prevent the popup scenario instead of the iframe scenario.

Reply:
This should only run in the redirect case.